docs/how-to-docs/cert-manager-upgrade-from-v0.8.md

```diff
@@ -0,0 +1,324 @@
+
+## Migration options
+
+There are 2 possible approaches for the migration of resources. The high-level steps for both approaches are expanded below.
+
+For a more detailed understanding of how the manifest files need to be updated, please refer to section `Updating cert-manager resources for v0.13.1` below.
+
+Option 1 below is strongly recommended as the approach.
+
+### Option 1 - renaming secrets
+
+By far the easiest and safest option is to amend the annotations and labels as described below **while at the same time also renaming the associated secrets**.
+
+Renaming the secret (changing the value of `secretName` in either `Ingress` or `Certificate` resources) will make sure that the same secret is not managed by 2 `Certificate` resources (the v0.8 certificate and its v0.13.1 counter-part) - whether those `Certificate` resources are part of your deployments or it's one of the resources managed automatically for you by cert-manager when it deals with `Ingress` annotations.
+
+To keep names consistent, you could for example add a `-cmio` suffix to all secret names (for `cert-manager.io`).
+
+- For `Ingress` resources:
+  - amend the `Ingress`'s annotations and labels
+  - change the value of `secretName`
+- For `Certificate` resources that are part of your deployments (e.g. to create a certificate that is mounted by an nginx sidecar for your main service):
+  - amend the `Certificate`'s annotations and labels
+  - change the value of `secretName`
+  - amend your deployment to mount the new secret (e.g. get the nginx sidecar to mount the new tls secret)
+- Deploy the changes
+- When you've checked that the service is functioning as intended, you can tidy up the old v0.8 cert-manager resources:
+  - delete any certificate resources still returned by `kubectl -n project get certificate` (back them up if they are not stored in git)
+  - delete any secrets associated with those old resources (again, back them up to be safe)
+- You can check that the certificate resources are valid by running `kubectl -n project get certificate.cert-manager.io`. The `READY` field for the resources should be `TRUE`. Note that it might take a short while (typically no more than about a minute) for the certificates to reach that `READY` state.
+
+The main draw-back of this approach is that the value for the new secret being created will need to be created from LetsEncrypt (unless using the platform culster issuer). This is usually quite quick, but could take up to around 2 minutes.
+
+If your deployments have several replicas and use a rolling update strategy, as they should, this is not usually an issue as the service will continue being available, although with a reduced capacity. As the deployment roll-out, users might experience a failed request that should be successful again on retry.
```

Comment:
I think this statement is a bit confusing. The user's workload fronted by Ingress shouldn't have any deployment changes (with these ingress annotation changes) and so remains operational.

With the secret name change on the ingress resource, there will temporarily be no valid cert for that ingress (whilst LE is performing the dns/http challenge), and so access to the endpoint is disrupted for that brief period of time whilst the challenge is completed and new cert/secret generated.